### PR TITLE
Rename AliasSeq to AliasTuple

### PR DESCRIPTION
**Pull request by JinShil**
See forum discussion http://forum.dlang.org/post/mnhfhs$2b9o$1@digitalmars.com

This is an alternative to https://github.com/D-Programming-Language/phobos/pull/3487

### CODE REVIEW

**Comment by andralex** [comment]
worksforme

**Comment by schuetzm** [comment]
+1

**Comment by Daniel-N** [comment]
+1

**Comment by schveiguy** [comment]
Yeah, I like this better too.

Ping @deadalnix @Hackerpilot @DmitryOlshansky @jmdavis (those who were in favor of, or discussed change to `AliasSeq`)

File: std/meta.d [hunk 1]
```diff
@@ -31,19 +31,19 @@
 module std.meta;
 
 /**
- * Creates a a sequence of zero or more aliases. This is most commonly
+ * Creates a list of zero or more aliases. This is most commonly
```
Comment (schveiguy) [inline std/meta.d]: Should this say "Creates a tuple" ?

**Comment by DmitryOlshansky** [comment]
I don't care that much. `AliasSplat` would be more unique but I'm OK with any of `AliasSeq`, `AliasTuple` too.

**Comment by jmdavis** [comment]
I really don't like calling this a tuple. That's part of the problem with `TypeTuple`. It really isn't a tuple, and the fact that we've called it a tuple has definitely caused confusion. It's similar to a tuple, but it isn't a tuple.

That being said, pretty much every name seems to suck on some level. `AliasTuple` is definitely a step up from `TypeTuple`, since it fixes the first half of the name, and every single suggestion for the second half has problems with it, because we're dealing with a fairly unique construct here. Maybe the word splat technically is more or less what we're looking for as has been suggested in the newsgroup, but I don't see how anyone could take that term seriously.

We simply don't have a good name for it. I think that `AliasTuple` is about as close as we're going to get to a name that actually describes what it does (except maybe for `AliasSplat`, but that's a horrible name), but it _is_ going to carry incorrect connotations thanks to baggage that comes with the term tuple. So, from that standpoint, it's probably best to either call it `AliasTuple` or pick a name that really says nothing about what it is so that no one makes any assumptions about it (like `iota`).

On the bright side, `AliasTuple` isn't as ugly-looking as `AliasSeq` and of a much more reasonable length than `AliasSequence`.

**Comment by Hackerpilot** [comment]
It's better than TypeTuple and Arguments, so sure. Let's use this one.

**Comment by schveiguy** [comment]
OK, doesn't seem like there's any concrete objections here.

**Comment by schveiguy** [comment]
Auto-merge toggled on

**Comment by schveiguy** [comment]
@MartinNowak I know that this probably should have gone into stable, but I think the original hadn't. Can you cherry pick this? I have no idea how to do that.

**Comment by deadalnix** [comment]
I'd like to object if that matter. This is not a list, (it is expected to have list of list working for instance).

I was never super happy with seq, but list is objectively worse. Discussion about it in the forum showed that this construct actually already havea name: splat.

If splat is not liked, then we should make our own word. Using list is no good. People where confused because this was not a tuple. People will be confused because this is not a list for the same reason: the word come with a baggage, and this baggage is creating confusion.

**Comment by JinShil** [comment]
> Can we please wait at least a full day with merging changes that are known to be controversial?

I  would prefer if most pull requests stayed open for at least a week (but not more than a month).  I think I still have local copies.  If you want me to revert this and resubmit it, please advise me of the proper procedure.

**Comment by schveiguy** [comment]
Sorry for pulling too quickly. I thought this was not too controversial. And we have an allowance here from Andrei. And the release is pending we don't want the wrong name to go out. The choice at this point seems so subjective, I think we should move on here.

At this point, not many liked Arguments. AliasSeq seemed doomed from W/A. Anything with List draws a veto from both Walter and Andrei. There simply aren't many choices. Fwiw, I think tuple is the most natural fit with existing D terminology

**Pull request merged**